modules/microshift-automation-example-rpm-systems.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * microshift/microshift-auto-recover-manual-backup.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-automation-example-rpm-systems_{context}"]
+= Automation integration example with systemd for RPM systems
+
+When the `microshift.service` enters a failed state, `systemd` starts the `microshift-auto-recovery.service` unit. This unit executes the auto-recovery restore process and restarts {microshift-short}.
+
+Use the following example to understand the automated recovery process for RPM systems using `systemd`.
+
+.Procedure
+
+. Create a directory for the `microshift.service` drop-in by running the following command:
++
+[source,terminal]
+----
+$ sudo mkdir -p /usr/lib/systemd/system/microshift.service.d
+----
+. Create the `10-auto-recovery.conf` file to instruct `systemd` to run `microshift-auto-recovery.service` when the `microshift.service` fails by running the following command:
++
+[source,terminal]
+----
+sudo tee /usr/lib/systemd/system/microshift.service.d/10-auto-recovery.conf > /dev/null <<'EOF'
+[Unit]
+OnFailure=microshift-auto-recovery.service
+EOF
+----
+. Create the `microshift-auto-recovery.service` file by running the following command:
++
+[source,terminal]
+----
+sudo tee /usr/lib/systemd/system/microshift-auto-recovery.service > /dev/null <<'EOF'
+[Unit]
+Description=MicroShift auto-recovery
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/microshift-auto-recovery
+
+[Install]
+WantedBy=multi-user.target
+EOF
+----
+. Create the `microshift-auto-recovery` script by running the following command:
```

Comment:
This is a bash script that needs to be placed in a file. It probably needs to be re-worked to include the sudo tee command.

Reply:
```
sudo tee /usr/bin/microshift-auto-recovery > /dev/null <<'EOF'
#!/usr/bin/env bash
set -xeuo pipefail

# If greenboot uses a non-default file for clearing boot_counter, use boot_success instead.
if grep -q  "/boot/grubenv" /usr/libexec/greenboot/greenboot-grub2-set-success; then
    if grub2-editenv - list | grep -q ^boot_success=0; then
        echo "Greenboot didn't decide the system is healthy after staging new deployment."
        echo "Quitting to not interfere with the process"
        exit 0
    fi
else
    if grub2-editenv - list | grep -q ^boot_counter=; then
        echo "Greenboot didn't decide the system is healthy after staging a new deployment."
        echo "Quitting to not interfere with the process"
        exit 0
    fi
fi

/usr/bin/microshift restore --auto-recovery /var/lib/microshift-auto-recovery
/usr/bin/systemctl reset-failed microshift
/usr/bin/systemctl start microshift

echo "DONE"
EOF
```

Reply:
done